src/stdio.jl

```diff
@@ -52,6 +65,63 @@ function watch_stream(rd::IO, name::AbstractString)
     end
 end
 
+function send_stdio(name)
+    if verbose::Bool && !haskey(task_local_storage(), :IJulia_task)
+        task_local_storage(:IJulia_task, "send $name task")
+    end
+    send_stream(name)
+end
+
+send_stdout(t::Timer) = send_stdio("stdout")
+send_stderr(t::Timer) = send_stdio("stderr")
+
+function send_stream(name::AbstractString)
+    buf = bufs[name]
+    if buf.size > 0
+        d = takebuf_array(buf)
+        n = num_utf8_trailing(d)
+        dextra = d[end-(n-1):end]
+        resize!(d, length(d) - n)
+        s = UTF8String(d)
+        if isvalid(s)
+            write(buf, dextra) # assume that the rest of the string will be written later
+            length(d) == 0 && return
+        else
+            # fallback: base64-encode non-UTF8 binary data
+            length(d) == 0 && return
```

Comment:
Good. I actually checked that `isvalid("") == true` earlier... too many hours at the screen today :)